lib/rouge/lexers/racket.rb

```diff
@@ -524,6 +525,13 @@ def self.builtins
         rule id, Name::Variable
       end
 
+      state :comment do
+        rule %r/[^|#]+/, Comment::Multiline
+        rule %r/\|#/, Comment::Multiline, :pop!
+        rule %r/#\|/, Comment::Multiline, :comment
+        rule %r/[|#]/, Comment::Multiline
```

Comment:
Which purpose does this line serve?

Reply:
This allows block comments to be nested. Otherwise the first `|#` will end the block. What's the right approach?

Reply:
Cool, I didn't know you were adding nested comments support. I provided nested samples from the docs in the comment.